Plugins/GeoModel/src/GeoModelToDetectorVolume.cpp

```diff
@@ -0,0 +1,186 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/GeoModel/GeoModelToDetectorVolume.hpp"
+
+#include "Acts/Detector/GeometryIdGenerator.hpp"
+#include "Acts/Detector/PortalGenerators.hpp"
+#include "Acts/Geometry/CuboidVolumeBounds.hpp"
+#include "Acts/Geometry/CutoutCylinderVolumeBounds.hpp"
+#include "Acts/Geometry/CylinderVolumeBounds.hpp"
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Geometry/TrapezoidVolumeBounds.hpp"
+#include "Acts/Navigation/DetectorVolumeFinders.hpp"
+#include "Acts/Navigation/InternalNavigation.hpp"
+
+#include <GeoModelKernel/GeoBox.h>
+#include <GeoModelKernel/GeoPcon.h>
+#include <GeoModelKernel/GeoShapeShift.h>
+#include <GeoModelKernel/GeoShapeSubtraction.h>
+#include <GeoModelKernel/GeoShapeUnion.h>
+#include <GeoModelKernel/GeoSimplePolygonBrep.h>
+#include <GeoModelKernel/GeoTrd.h>
+#include <GeoModelKernel/GeoTube.h>
+#include <GeoModelKernel/GeoTubs.h>
+
+namespace Acts {
+namespace GeoModel {
+std::shared_ptr<Experimental::DetectorVolume> convertVolume(
```

Comment:
Is it in-scope to factorize these functions to produce the `VolumeBounds` separately from the detector volumes? 
Ideally, and looking forward to Gen3, I think the best approach would be for these functions to produce standard `Acts::Volume`s, instead of `DetectorVolumes`. 
The Portal generation and navigation delegates are hard-coded anyway, so the conversion from `Acts::Volume` to `DetectorVolume` should be easy. This would allow this code to continue to be useful, rather than being superceded in the near future.

Reply:
The (verry convenient) way in which I did the conversion is to pass the GeoShape to the ToDetecotorVolume which passes it to a different converter. Should I then change all of that to get Acts::Volume as return? Can/should Acts::Volume also contain the surfaces?

Reply:
I would prefer to do that in a future refactor pr